lib/screens/customer_product_detail_page.dart

```diff
@@ -312,17 +362,23 @@ class _CustomerProductDetailPageState extends State<CustomerProductDetailPage> {
   }
 
   Future<void> _addToCart() async {
-    var customer = Customer.fromSnapshot(await db
-        .collection(constants.DBCollections.users)
-        .document(currentSignedInUser.email)
-        .get());
-
-    _updateCartModel(customer);
-
-    await db
-        .collection(constants.DBCollections.users)
-        .document(currentSignedInUser.email)
-        .updateData(customer.toMap());
+    if (_selectedItemColor != null && _selectedItemSize != null) {
```

Comment:
I think this check is valid only for `stockType` = `sizeAndColor`. 

You can add a helper method like this:
```
bool _canAddToCart() {
  if (_product.stock.stockType == StockType.sizeAndColor) {
    return _selectedItemColor != null && _selectedItemSize != null;
  } else if (_product.stock.stockType == StockType.sizeOnly) {
    return _selectedItemSize != null;
  } else if (_product.stock.stockType == StockType.colorOnly) {
    return _selectedItemColor != null;
  }
  return true;
}
```
Then, you can use this method in the `GestureDetector` widget of `_buildAddToCartButton`:
```
if (_canAddToCart()) {
  _addToCart();
} else {
  setState(() {
    _showAddToCartFailedMessage = true;
  });
}
```
This way, you don't have to modify the `_addToCart` method.